spikeinterface/sortingcomponents/motion_estimation.py

```diff
@@ -404,34 +439,41 @@ def get_windows(rigid, bin_um, contact_pos, spatial_bin_edges, margin_um, win_st
     Here by default we use gaussian window.
 
     """
-    bin_centers = spatial_bin_edges[:-1] + bin_um /2.
+    bin_centers = spatial_bin_edges[:-1] + bin_um / 2.
     n = bin_centers.size
 
     if rigid:
-        # win_shape = 'rect' is forced
+        # win_shape = 'rect' is forced
         non_rigid_windows = [np.ones(n, dtype='float64')]
         middle = (spatial_bin_edges[0] + spatial_bin_edges[-1]) / 2.
         non_rigid_window_centers = np.array([middle])
     else:
-        assert win_sigma_um > win_step_um, f'win_sigma_um too low {win_sigma_um} compared to win_step_um {win_step_um}'
+        assert win_sigma_um >= win_step_um, f'win_sigma_um too low {win_sigma_um} compared to win_step_um {win_step_um}'
 
         min_ = np.min(contact_pos) - margin_um
         max_ = np.max(contact_pos) + margin_um
         num_non_rigid_windows = int((max_ - min_) // win_step_um)
-        border = ((max_ - min_)  %  win_step_um) / 2
+        border = ((max_ - min_) % win_step_um) / 2
         non_rigid_window_centers = np.arange(num_non_rigid_windows + 1) * win_step_um + min_ + border
         non_rigid_windows = []
-        
+
         for win_center in non_rigid_window_centers:
             if win_shape == 'gaussian':
                 win = np.exp(-(bin_centers - win_center) ** 2 / (win_sigma_um ** 2))
             elif win_shape == 'rect':
                 win = np.abs(bin_centers - win_center) < (win_sigma_um / 2.)
                 win = win.astype('float64')
+            elif win_shape == 'triangle':
+                center_dist = np.abs(bin_centers - win_center)
+                in_window = center_dist <= (win_sigma_um / 2.)
+                win = -center_dist
+                win[~in_window] = 0
+                win[in_window] -= win[in_window].min()
+                win[in_window] /= win[in_window].max()
```

Comment:
Finally the trend is on triangle ??

Reply:
Haha, I like them, and you'll like this -- with `win_shape == 'triangle'` and `win_sigma_um == 2 * win_step_um`, then `upsample_to_histogram_bin` leads to linear interpolation of the motion estimate between bins :)